samples/notesync/NoteSync.html

```diff
@@ -0,0 +1,300 @@
+      </div>
+    </div>
+  </div>
+</script>
+
+<script>
+// -------------------------------------------------- The Models ---------------------------------------------------- //
+// The Note Model
+app.models.Note = Force.SObject.extend({
+    sobjectType: "ContentNote",
+    fieldlist: ["Id", "Title", "Content", "LastModifiedDate"],
+    cache: function() { return app.cache;},
+    cacheMode: Force.CACHE_MODE.CACHE_ONLY
+});
+
+// The NoteCollection Model
+app.models.NoteCollection = Force.SObjectCollection.extend({
+    model: app.models.Note,
+    cache: function() { return app.cache;},
+
+    getCriteria: function() {
+        return this.key;
+    },
+
+    setCriteria: function(key) {
+        this.key = key;
+        this.config = {type:"cache", 
+                       cacheQuery:{queryType:"smart", 
+                                   smartSql:"SELECT {notes:_soup}"
+                                   + " FROM {notes}"
+                                   + " WHERE {notes:Title} like '" + key + "%'"
+                                   + " ORDER BY {notes:Title} "
+                                   + " LIMIT 25 ", 
+                                   pageSize:25}
+                      };
+    }
+});
+
+
+// -------------------------------------------------- The Views ---------------------------------------------------- //
+
+app.views.SearchPage = Backbone.View.extend({
+
+    template: _.template($("#search-page").html()),
+
+    events: {
+        "keyup .search-key": "search",
+        "click .syncdown": "syncDown",
+        "click .syncup": "syncUp",
+        "click .inspect": "inspect"
+    },
+
+    initialize: function() {
+        var that = this;
+        this.fieldlist = app.models.Note.prototype.fieldlist;
+        this.listView = new app.views.NoteListView({model: this.model});
+        document.addEventListener("sync", function() { 
+            that.handleSyncUpdate(event.detail);
+        });
+    },
+
+    handleSyncUpdate: function(sync) {
+        var progress =  (typeof sync.progress === "number" && sync.progress < 100) ? " (" + Math.round(sync.progress) + "%)" : "";
+
+        if (sync.type === "syncDown") {
+            this.lastSoqlSyncDownId = sync.syncId;
+            $(".syncdown").html("Sync " + progress);
+        }
+        if (sync.type === "syncUp") {
+            $(".syncup").html("Sync Up" + progress);
+        }
+        if (sync.status === "DONE") { this.search(); }
+    },
+
+    render: function(eventName) {
+        $(this.el).html(this.template());
+        $(".search-key", this.el).val(this.model.getCriteria());
+        this.listView.setElement($("ul", this.el)).render();
+        return this;
+    },
+
+    syncDown: function() {
+        var target = {type:"custom", androidImpl:"com.salesforce.samples.notesync.ContentSoqlSyncTarget", query:"SELECT " + this.fieldlist.join(",") + " FROM ContentNote LIMIT 1000"};
```

Comment:
Notice the custom sync target with the androidImpl attribute to specify the implementation (I'll add a iosImpl once I have support for custom sync target on iOS).
I'll send a pull request later for the Android repo with the ContentSoqlSyncTarget class.